server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/webhook/WebhookPayloadFactoryImpl.java

```diff
@@ -73,8 +73,13 @@ private void writeServer(JsonWriter writer) {
   }
 
   private static void writeDates(JsonWriter writer, PostProjectAnalysisTask.ProjectAnalysis analysis, System2 system2) {
-    analysis.getAnalysisDate().ifPresent(date -> writer.propDateTime("analysedAt", date));
-    writer.propDateTime("changedAt", analysis.getAnalysisDate().orElseGet(() -> new Date(system2.now())));
+    analysis.getAnalysis().ifPresent(a -> writer.propDateTime("analysedAt", a.getDate()));
+    if (analysis.getAnalysis().isPresent()) {
+      writer.propDateTime("changedAt", analysis.getAnalysis().get().getDate());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)

Reply:
FP